infra/modules/azure_naming_convention/locals.tf

```diff
@@ -7,10 +7,81 @@ locals {
     "spaincentral"       = "spc"
   }
 
-  location_short = lookup(local.location_map, var.environment.location, "neu")
-  project        = "${var.environment.prefix}-${var.environment.env_short}-${local.location_short}"
-  domain         = var.environment.domain == null ? "-" : "-${var.environment.domain}-"
+  configurations = {
+    for env in var.environments : join("", [
+      "${env.prefix}-${env.env_short}-${lookup(local.location_map, env.location, "neu")}",
+      env.domain == null ? "-" : "-${env.domain}-",
+      env.app_name
+      ]) => {
+      prefix          = env.prefix
+      env_short       = env.env_short
+      location        = env.location
+      domain          = env.domain
+      app_name        = env.app_name
+      instance_number = env.instance_number
+      project         = "${env.prefix}-${env.env_short}-${lookup(local.location_map, env.location, "neu")}"
+      app_suffix      = [for i in range(env.instance_number) : format("%02d", i + 1)]
```

Comment:
What does this line?

Reply:
app_suffix? Now `instance_number` is a number between 1 and 99, with this `for` it generates a list with all suffixes in the classical form 01, 02, ...
Example, if instance number is 3, app_suffix is a list ["01", "02", "03"]
In this way we don't have the only 01 case as before

Reply:
what about adding validation for instance number > 1 <= 99?

Reply:
it's already present in `variables.tf` file

```
  validation {
    condition     = alltrue([for env in var.environments : env.instance_number > 0 && env.instance_number < 100])
    error_message = "The variable \"instance_number\" only accepts values in the range [1-99]."
  }
```